did/key/key_vectors_test.go

```diff
@@ -0,0 +1,112 @@
+package key
+
+import (
+	"embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	//go:embed testdata
+	testData embed.FS
+)
+
+const (
+	Ed25519X25519TestVector string = "ed25519-x25519.json"
+	X25519TestVector        string = "x25519.json"
+	NISTCurvesTestVector    string = "nist-curves.json"
+	RSATestVector           string = "rsa.json"
+	SECP256k1TestVector     string = "secp256k1.json"
+)
+
+func TestKnownTestVectors(t *testing.T) {
+	tests := []struct {
+		name     string
+		testFile string
+	}{
+		{
+			name:     "Ed25519/X25519",
+			testFile: Ed25519X25519TestVector,
+		},
+		{
+			name:     "X25519",
+			testFile: X25519TestVector,
+		},
+		{
+			name:     "NIST Curves",
+			testFile: NISTCurvesTestVector,
+		},
+		{
+			name:     "RSA",
+			testFile: RSATestVector,
+		},
+		{
+			name:     "secp256k1",
+			testFile: SECP256k1TestVector,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			testVector := retrieveTestVector(tt, test.testFile)
+			for id, vector := range testVector {
+				didKey := DIDKey(id)
+
+				var pubKeyFormatOption Option
+				if vector.DIDDocument.VerificationMethod[0].Type == cryptosuite.JSONWebKey2020Type {
+					pubKeyFormatOption = PublicKeyFormatJSONWebKey2020
+				} else {
+					pubKeyFormatOption = PublicKeyFormatMultibase
+				}
+
+				var enableEncryptionKeyDerivationOption Option
+				if len(vector.DIDDocument.VerificationMethod) < 2 {
+					enableEncryptionKeyDerivationOption = DisableEncryptionKeyDerivation
+				}
+				// note: all other test vectors have this flag set
+				if test.name != "Ed25519/X25519" && test.name != "X25519" {
+					enableEncryptionKeyDerivationOption = EnableEncryptionKeyDerivation
+				}
+				didDoc, err := didKey.Expand(pubKeyFormatOption, enableEncryptionKeyDerivationOption)
```

Comment:
Seems like these should be passed in as fields in the test table.

Reply:
it can't be because it depends on the vector itself - not the file, since each file contains a mix of options